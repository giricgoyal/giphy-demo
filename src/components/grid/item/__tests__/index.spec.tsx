import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom'

import GridItem from '../index'

describe('<GridItem />', () => {
    let container, getByTestId
    const onClickHander = jest.fn()

    beforeEach(() => {
        ;({ container, getByTestId } = render(
            <GridItem id="bugs-bunny" onClick={onClickHander}>
                <div>Bugs Bunny</div>
            </GridItem>,
        ))
    })

    test('should match snapshot', () => {
        expect(container).toMatchSnapshot()
    })

    describe('onClick', () => {
        describe('when clicked', () => {
            beforeEach(() => {
                fireEvent.click(getByTestId('grid-item'))
            })

            test('should call the onClickHandler', () => {
                expect(onClickHander).toHaveBeenCalledWith('bugs-bunny')
            })
        })
    })
})
