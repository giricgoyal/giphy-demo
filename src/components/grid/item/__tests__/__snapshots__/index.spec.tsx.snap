// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridItem /> should match snapshot 1`] = `
<div>
  <div
    class="grid__item cursor-pointer"
    data-testid="grid-item"
  >
    <div>
      Bugs Bunny
    </div>
  </div>
</div>
`;
